### PR TITLE
Declare rules_python deps

### PR DESCRIPTION
**Pull request by katre**
[Previous text moved to #179]

Add correct lxml dep in rules_python setup, fixing #179.

### CODE REVIEW

**Comment by katre** [comment]
Okay, after some work and updates I at least have consistent failures: without bzlmod, enforce_min_sdk_floor can't run because the python toolchain can't find the binary dependencies on lxml (this is an assumption but seems justified after checking similar error reports in other programs).

I don't know why bzlmod works here, and what it's doing differently.

**Comment by katre** [comment]
Okay, that solved it for linux, mac intel and mac arm apparently need separate vendored_requirements files because of the interpreter difference. I'll try a bit more and see if I can poke this.

**Comment by ted-xie** [comment]
Thanks for looking into this!

> Managed to set up a docker image that mimics CI, and it looks like the (successful) bzlmod version is using python 3.11, and the (unsuccessful) non-bzlmod version is using python 3.8:

wat, there are two different versions of python used in two different build pipelines in BazelCI?

**Comment by katre** [comment]
Yes, the Mac workers and the Ubuntu workers have different versions installed.

Ideally, we wouldn't have system python on either, and just use a hermetic python toolchain for everything.

To summarize the problem for rules_android:
1. To use a hermetic python toolchain is easy with bzlmod
    1. We don't publish a bzlmod version of rules_android for users, even though it's possible to build inside rules_android with bzlmod
2. To use a hermetic python toolchain without bzlmod, we'd need a third-stage WORKSPACE macro (after `rules_android_prereqs` and `rules_android_workspace`
   1. It's possible to vendor the needed deps files, except that they have architecture dependencies, so can't work for us.

File: py_support/requirements.in [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# If this changes, regenerate requirements_lock.txt:
+# $ bazel run --enable_bzlmod=false //py_support:requirements.update
+
+--extra-index-url https://pypi.python.org/simple/
+
+lxml
```
[the remark this answers is not in the document]
Comment (katre) [reply py_support/requirements.in:6]: I believe it's locked to the latest version when the lockfile was last updated.

It'd probably be better to explicitly specify a version, but that's down the chain after "make the thing work cross-platform".

**Pull request merged**